types/office-js-preview/index.d.ts

```diff
@@ -18491,14 +18547,266 @@ declare namespace Office {
+         * @param options - An object literal that contains one or more of the following properties:-
+         *        `asyncContext`: Developers can provide any object they wish to access in the callback function.
+         * @param callback - Optional. When the method completes, the function passed in the `callback` parameter is called with a single parameter, `asyncResult`,
+         *                 which is an `Office.AsyncResult` object.
+         * 
+         * @beta
+         */
+        setAsync(sensitivityLabel: string | SensitivityLabelDetails, options: Office.AsyncContextOptions, callback?: (asyncResult: Office.AsyncResult<void>) => void): void;
+        /**
+         * Applies the specified sensitivity label to the message or appointment being composed.
+         * 
+         * @remarks
+         * [Api set: Mailbox preview]
+         * 
+         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/understanding-outlook-add-in-permissions | Minimum permission level}**: **read/write item**
+         *
+         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/outlook-add-ins-overview#extension-points | Applicable Outlook mode}**: Compose
+         * 
+         * **Tip**: To determine the sensitivity labels available for use, call the `Office.context.sensitivityLabelsCatalog.getAsync` method.
+         * 
+         * @param sensitivityLabel - The sensitivity label to be applied to the message or appointment being composed. The parameter value can be a sensitivity label's
+         *                         unique identifier (GUID) or a {@link Office.SensitivityLabelDetails | SensitivityLabelDetails} object.
+         * @param callback - Optional. When the method completes, the function passed in the `callback` parameter is called with a single parameter, `asyncResult`,
+         *                 which is an `Office.AsyncResult` object.
+         * 
+         * @beta
+         */
+        setAsync(sensitivityLabel: string | SensitivityLabelDetails, callback?: (asyncResult: Office.AsyncResult<void>) => void): void;
+    }
+    /**
+     * Provides methods to check the {@link https://learn.microsoft.com/microsoft-365/compliance/sensitivity-labels | sensitivity labels catalog} status in Outlook and
+     * retrieve all available sensitivity labels if the catalog is enabled.
+     * 
+     * @remarks
+     * [Api set: Mailbox preview]
+     * 
+     * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/understanding-outlook-add-in-permissions | Minimum permission level}**: **read/write item**
+     *
+     * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/outlook-add-ins-overview#extension-points | Applicable Outlook mode}**: Compose
+     * 
+     * @beta
+     */
+    export interface SensitivityLabelsCatalog {
+        /**
+         * Gets all the sensitivity labels enabled in Outlook.
```

Comment:
Consideration for clarity.
```suggestion
         * Gets all the sensitivity labels that are enabled in Outlook.
```